packages/compass-aggregations/src/components/stage-wizard/index.tsx

```diff
@@ -145,15 +160,37 @@ type WizardOwnProps = {
 
 export default connect(
   (state: RootState, ownProps: WizardOwnProps) => {
-    const wizard = state.pipelineBuilder.stageEditor.stages[
-      ownProps.index
-    ] as Wizard;
+    const {
+      autoPreview,
+      fields: initialFields,
+      pipelineBuilder: {
+        stageEditor: { stages },
+      },
+    } = state;
+
+    const wizard = stages[ownProps.index] as Wizard;
+
+    const previousStage = stages
+      .slice(0, ownProps.index)
+      .reverse()
+      .find((x) => x.type === 'stage' && !x.disabled) as StoreStage | undefined;
```

Comment:
```suggestion
      .find((x): x is StoreStage => x.type === 'stage' && !x.disabled);
```
This will take care of both.